services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,16 +2,59 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import taskcluster from 'taskcluster-client';
+import { decryptToken } from './decryptToken.js';
+import { ErrorReply } from 'taskcluster-lib-api';
 
-export default ({ server, schema, context, path }) =>
+// TODO: Check for expiration of access token when the websocket connection is active
+export default ({ cfg, server, schema, context, path }) =>
   SubscriptionServer.create(
     {
       schema,
       execute,
       subscribe,
-      onConnect(params, socket) {
-        return new Promise(resolve => {
-          credentials()(socket.upgradeReq, {}, resolve);
+      async onConnect(params, socket) {
+
+        setTimeout(() => {
+          socket.close();
```

Comment:
does it  need a check if socket is closed already? 
Maybe we need to handle this to avoid throwing unnecessary errors. 

Also, maybe it would be good alternatively to listen to socket disconnect event and clear this timeout? 

Just a thought